### PR TITLE
Reimplement using GStreamer and Rust

### PR DESCRIPTION
**Pull request by otavio**
This PR reimplements the EasySplash using GStreamer as backend. This is the first preview and likely will be changed prior it is released.

To run this, you can use:

```
ES_RUNTIME_DIR=/tmp/es GST_DEBUG=1 RUST_LOG=trace cargo run -- open data/ossystems-demo
```

and to send commands to interrupt the animation (current animation is not interruptable) you can use:

```
ES_RUNTIME_DIR=/tmp/es RUST_LOG=trace cargo run -- client --stop
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,11 @@
...
-The frames are divided in parts. The archive format is documented [here](doc/Animation-Structure.md)
-
-It is possible to run EasySplash in X11 and on the PC. This is useful for debugging purposes
-and for trying out new animations on the PC before uploading it to embedded devices.
-
-EasySplash can run in realtime or non-realtime mode. realtime mode means the animaton progresses
-in realtime, without user interaction. In non-realtime mode, the animation progresses only by
-sending progress updates over easysplashctl. The latter is useful if a progress value can be
-determined while something is loading, for example.
+It is possible to run EasySplash in X11 and on the PC. This is useful for debugging purposes and for
```
Comment (Jonathas-Conceicao) [inline README.md:7]: "possible to run [..] in X11 and on the PC" seems a bit redundant here.

File: src/main.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+struct Client {
+    /// stop the render as soon as possible
+    #[argh(switch)]
+    stop: bool,
+}
+
+#[async_std::main]
+async fn main() -> Result<(), anyhow::Error> {
+    // Initilize the logger (default to info level)
+    env_logger::from_env(Env::default().default_filter_or("info")).init();
```
Comment (Jonathas-Conceicao) [inline src/main.rs:53]: Having the log controlled by the command line options itself would be more user-friendly.

File: src/message.rs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    fn from(msg: Message) -> Self {
+        match msg {
+            Message::Interrupt => 0x1,
+        }
+    }
+}
+
+async fn unix_socket_file(create_missing: bool) -> Result<PathBuf, io::Error> {
+    let path = PathBuf::from(
+        env::var("ES_RUNTIME_DIR").unwrap_or_else(|_| String::from("/run/easysplash")),
```
Comment (Jonathas-Conceicao) [inline src/message.rs:55]: This would be better as a command line option.

Comment (Jonathas-Conceicao) [inline src/message.rs:55]: Having the default path set to `/run/easysplash` might not be ideal here. When running with a usual linux distro the normal user might not have the write permissions for that path, which would make first time users get an error right on the first run.

File: README.md [hunk 4]
```diff
@@ -28,150 +20,67 @@ There are two examples, which may be used as reference:
 Requirements
```
Comment (Jonathas-Conceicao) [inline README.md:20]: Can't add this note on the right lines. I think that the links above should be updated.

File: README.md (hunk 1 above)
Comment (otavio) [reply README.md:7]: Done.

File: src/main.rs (hunk 2 above)
Comment (otavio) [reply src/main.rs:53]: I converted it to use `simple_logging` and added a command line for both open and client sub commands.

File: src/message.rs (hunk 3 above)
Comment (otavio) [reply src/message.rs]: Done.

Comment (otavio) [reply src/message.rs]: Done.

**Comment by otavio** [comment]
@Jonathas-Conceicao I'll rework the examples on README but please review the rest of the PR.

File: README.md (hunk 4 above)
Comment (otavio) [reply README.md:20]: Ok, I added the second animation and added the links on README.

**Pull request merged**